src/__experimental__/components/checkbox/checkbox-group.component.js

```diff
@@ -2,42 +2,73 @@ import React from "react";
 import PropTypes from "prop-types";
 import tagComponent from "../../../utils/helpers/tags";
 import { StyledCheckboxGroup } from "./checkbox.style";
-import FormField from "../form-field";
-import { InputGroupBehaviour } from "../../../__internal__/input-behaviour";
+import Fieldset from "../../../__internal__/fieldset";
 
 const CheckboxGroup = (props) => {
-  const { children, groupName, error, warning, info } = props;
+  const {
+    children,
+    groupName,
+    legend,
+    error,
+    warning,
+    info,
+    mb,
+    required,
+    legendInline,
+    legendWidth,
+    legendAlign,
+    legendSpacing,
+  } = props;
 
   const groupLabelId = `${groupName}-label`;
 
   return (
-    <InputGroupBehaviour>
+    <Fieldset
+      role="group"
```

Comment:
After I took a look at this https://www.w3.org/WAI/tutorials/forms/grouping/ I think we don't need to use `role="group"` on this `Fieldset` element

Reply:
actually we don't need to use any role on this element and it's descendant, I'll remove both of them